Common/Orders/MarketOrder.cs

```diff
@@ -42,6 +42,49 @@ public override decimal Value
         }
 
         /// <summary>
+        /// Create update request for pending orders. Null values will be ignored.
+        /// </summary>
+        public UpdateOrderRequest CreateUpdateRequest(int? quantity = null, string tag = null)
+        {
+            return new UpdateOrderRequest
+            {
+                Id = Guid.NewGuid(),
+                OrderId = Id,
+                Created = DateTime.Now,
+                Quantity = quantity ?? Quantity,
+                Tag = tag ?? Tag
+            };
+        }
+
+        /// <summary>
+        /// Create submit request.
+        /// </summary>
+        public static SubmitOrderRequest CreateSubmitRequest(SecurityType securityType, string symbol, int quantity, DateTime time, decimal price = 0, string tag = null)
+        {
```

Comment:
Very well. So I should merge your upstream/master with my branch and see
what happens?

On Thu, Jul 2, 2015 at 5:23 PM, Michael <notifications@github.com> wrote:

> In Common/Orders/MarketOrder.cs
> <https://github.com/QuantConnect/Lean/pull/134#discussion_r33824281>:
>
> > +        {
> > +            return new UpdateOrderRequest
> > +            {
> > +                Id = Guid.NewGuid(),
> > +                OrderId = Id,
> > +                Created = DateTime.Now,
> > +                Quantity = quantity ?? Quantity,
> > +                Tag = tag ?? Tag
> > +            };
> > +        }
> > +
> > +        /// <summary>
> > +        /// Create submit request.
> > +        /// </summary>
> > +        public static SubmitOrderRequest CreateSubmitRequest(SecurityType securityType, string symbol, int quantity, DateTime time, decimal price = 0, string tag = null)
> > +        {
>
> The MarketOrder type requires the price, the others should have price set
> to zero. With that being said, @jaredbroad <https://github.com/jaredbroad>
> and I spoke about changing how this works in the future. Currently the
> price is needed for the Value property on Order during the call to
> GetSufficientCapitalForOrder. The new idea is to have an method such as:
>
> public abstract decimal GetOrderValue(decimal marketPrice)
>
> Which all orders will implement, for now let's just leave it alone though,
> we'll come back to this in due time.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/QuantConnect/Lean/pull/134/files#r33824281>.
>